### DISCUSSION TITLE
Virtual slides with dynamic data fetched from data source in React

### DISCUSSION THREAD

**Discussion by simplecommerce** [nolimits4web/swiper#4740] [Q&A]
Hi,

I am using React and I am trying to find an example on how to load slides dynamically after fetching them from a query from my data source.

I was thinking I would have to use virtual slides.

But your example seems to be automated and doesn't allow me to use `renderExternal` to fetch the data.

This is the example I am looking at: https://swiperjs.com/react#virtual-slides

What I want is to load by default X amount of slides, but when they click next, if they reach near the end, it will fetch the remote to see if there is additional data and update the slides.

I might be missing something or not understanding it correctly.

Hope you can point me out in the right direction.

Thanks!

**Comment by duoluodexiaoxiaoyuan** [comment]
you question is don't  know  how render new data  after previous data is the best behind.  me too confused

**Reply by simplecommerce** [reply]
> you question is don't know how render new data after previous data is the best behind. me too confused

I don't remember my issue here, but in my project I used `reachEnd` event to fire my query to load additional slides.

**Reply by duoluodexiaoxiaoyuan** [reply]
> > you question is don't know how render new data after previous data is the best behind. me too confused
> 
> I don't remember my issue here, but in my project I used `reachEnd` event to fire my query to load additional slides.

Now  i meet a question.  

For  example ,  i  need release carousel  pictures(these  pictures  accors api  return),  i don’t   how render new data after previous data is the best behind. 

I  google  don't find good solution plan.  Are you know.

Accors  to use Swiper,I found that it made some processing.  

For example,  once API  return [1,2,3],   twice API return [1,2,3,4].  you can find it's  no interval. But  twice  API reutn [9,10],  it's  immediately show  twice data,  whether once  data  carousel  is the last.

**Reply by simplecommerce** [reply]
> > > you question is don't know how render new data after previous data is the best behind. me too confused
> > 
> > 
> > I don't remember my issue here, but in my project I used `reachEnd` event to fire my query to load additional slides.
> 
> Now i meet a question.
> 
> For example , i need release carousel pictures(these pictures accors api return), i don’t how render new data after previous data is the best behind.
> 
> I google don't find good solution plan. Are you know.
> 
> Accors to use Swiper,I found that it made some processing.
> 
> For example, once API return [1,2,3], twice API return [1,2,3,4]. you can find it's no interval. But twice API reutn [9,10], it's immediately show twice data, whether once data carousel is the last.

I am sorry, I have a hard time understanding what you said without a working example.
